data/pigui/modules/info-view/01-ship-info.lua

```diff
@@ -1,33 +1,34 @@
 -- Copyright © 2008-2024 Pioneer Developers. See AUTHORS.txt for details
 -- Licensed under the terms of the GPL v3. See licenses/GPL-3.txt
 
-local Equipment = require 'Equipment'
 local Event = require 'Event'
 local Game = require 'Game'
 local Lang = require 'Lang'
 local ShipDef = require 'ShipDef'
 local InfoView = require 'pigui.views.info-view'
+local Passengers = require 'Passengers'
 local Vector2 = Vector2
 
 local ui = require 'pigui'
 local l = Lang.GetResource("ui-core")
 
-local fonts = ui.fonts
 local textTable = require 'pigui.libs.text-table'
 
 local equipmentWidget = require 'pigui.libs.ship-equipment'.New("ShipInfo")
 
 local function shipStats()
 	local player = Game.player
+	local equipSet = player:GetComponent("EquipSet")
 
 	-- Taken directly from ShipInfo.lua.
 	local shipDef       =  ShipDef[player.shipId]
 	local shipLabel     =  player:GetLabel()
-	local hyperdrive    =  table.unpack(player:GetEquip("engine"))
-	local frontWeapon   =  table.unpack(player:GetEquip("laser_front"))
-	local rearWeapon    =  table.unpack(player:GetEquip("laser_rear"))
-	local cabinEmpty    =  player:CountEquip(Equipment.misc.cabin, "cabin")
-	local cabinOccupied =  player:CountEquip(Equipment.misc.cabin_occupied, "cabin")
+	local hyperdrive    =  equipSet:GetInstalledOfType("hyperdrive")[1]
+	local frontWeapon   =  equipSet:GetInstalledOfType("weapon")[1]
+	local rearWeapon    =  equipSet:GetInstalledOfType("weapon")[2]
```

Comment:
Is this correct? Shouldn't you be searching for the weapon installed in the rear slot now instead of just returning the second weapon?

Reply:
This is patched solely for the purpose of making the code run. We currently don't provide any "rear weapon" slots on most ships, and the functionality to have a rear-firing weapon will be getting addressed in the GunManager branch.

TL;DR: this is a legacy expectation and will be migrated to a newer system.